source/fortran/init_module.f90

```diff
@@ -176,6 +176,38 @@ subroutine init
 
    end subroutine init
 
+   subroutine open_idempotence_files
+      ! Open new output file and mfile to write output to
+      ! This is used when checking model evaluation idempotence, to avoid
+      ! polluting the final output file and mfile with intermediate result checks
+      use global_variables, only: output_prefix
+      use constants, only: nout, mfile
+      implicit none
+
+      ! Close existing output file and mfile (could be original out and mfiles
+      ! or idem scratch files)
+      close(unit = nout)
+      close(unit = mfile)
+      ! Open scratch files with same units
+      open(unit=nout, file=trim(output_prefix)//'IDEM_OUT.DAT', action='write', status='replace')
+      open(unit=mfile, file=trim(output_prefix)//'IDEM_MFILE.DAT', action='write', status='replace')
+   end subroutine open_idempotence_files
+
+   subroutine close_idempotence_files
+      ! Revert back to writing to original OUT.DAT and MFILE.DAT, after
```

Comment:
I know what you're saying and doing, but it might be worth being a bit more explicit:
* Closing the intermediate idempotence-check files, deleting them in the process
* Re-opening the actual output files ready to write the final data

Reply:
Made more explicit.